### ISSUE TITLE
local variable 'display_status' referenced before assignment

### ISSUE THREAD

**Issue by barbacbd** [ansible/ansible_collections_azure#1193]
##### SUMMARY
Running a virtual machine scaleup, and the playbook's task failed with 
```
line 468, in serialize_vm\nUnboundLocalError: local variable 'display_status' referenced before assignment\n",
```

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

azure_rm_virtualmachine_info.py

##### ANSIBLE VERSION

```paste below
  config file = None
  configured module search path = ['/home/bbarbach/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/bbarbach/venv/lib64/python3.9/site-packages/ansible
  ansible collection location = /home/bbarbach/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/bbarbach/venv/bin/ansible
  python version = 3.9.6 (default, Aug 11 2021, 06:39:25) [GCC 8.5.0 20210514 (Red Hat 8.5.0-3)] (/home/bbarbach/venv/bin/python3.9)
  jinja version = 3.1.2
  libyaml = True
```

##### COLLECTION VERSION

```paste below
azure.azcollection 1.15.0 
```

##### CONFIGURATION

```paste below
CONFIG_FILE() = None
```

##### OS / ENVIRONMENT

RHEL 8.7

##### STEPS TO REPRODUCE

```yaml
---
- name: Retrieve platform VM list
  azure.azcollection.azure_rm_virtualmachine_info:
    resource_group: "{{ azure_resource_group }}"
    client_id: "{{ azure_client_id }}"
    subscription_id: "{{ azure_subscription_id }}"
    tenant: "{{ azure_tenant_id }}"
    secret: "{{ azure_secret }}"
  register: azure_vms
  run_once: true

- name: Set azure vm to most recent image
  set_fact:
    azure_vm: "{{ azure_vms.vms[-1].name }}"
  run_once: true

- name: Create machineset_name
  set_fact:
    machineset_name: "{{ machineset_obj.metadata.name ~ '-' ~ distribution ~ '-' ~ distribution_version }}"

# stop the playbook here
#- ansible.builtin.meta: end_play

- name: Update machineset definition
  set_fact:
    machineset: "{{ machineset_obj | combine(dict_edit, recursive=True) }}"
  vars:
    dict_edit:
      metadata:
        name: "{{ machineset_name }}"
        resourceVersion: ""
      spec:
        selector:
          matchLabels:
            machine.openshift.io/cluster-api-machineset: "{{ machineset_name }}"
        template:
          metadata:
            labels:
              machine.openshift.io/cluster-api-machineset: "{{ machineset_name }}"
          spec:
            providerSpec:
              value:
                image:
                  offer: RHEL
                  resourceID: ""
                  publisher: RedHat
                  sku: 8-lvm-gen2
                  version: latest

- name: Import machineset definition
  command: >
    oc apply -f -
    --kubeconfig={{ kubeconfig }}
  register: oc_apply
  args:
    stdin: "{{ machineset | to_yaml }}"
  changed_when:
  - ('created' in oc_apply.stdout) or
    ('configured' in oc_apply.stdout)

- name: Get machines in the machineset
  command: >
    oc get machine
    --kubeconfig={{ kubeconfig }}
    --namespace=openshift-machine-api
    --selector='machine.openshift.io/cluster-api-machineset={{ machineset_name }}'
    --output=json
  register: oc_get_machine
  until:
  - oc_get_machine.stdout != ''
  - (oc_get_machine.stdout | from_json)['items'] is defined
  - (oc_get_machine.stdout | from_json)['items'] | length > 0
  changed_when: false

- name: Create list of machines
  set_fact:
    worker_machines: "{{ (oc_get_machine.stdout | from_json)['items'] | map(attribute='metadata.name') | list }}"

- name: Wait for machines to be provisioned
  command: >
    oc get machine {{ item }}
    --kubeconfig={{ kubeconfig }}
    --namespace=openshift-machine-api
    --output=json
  loop: "{{ worker_machines }}"
  register: new_machine
  until:
  - new_machine.stdout != ''
  - (new_machine.stdout | from_json).status is defined
  - (new_machine.stdout | from_json).status.phase == 'Provisioned'
  retries: 36
  delay: 5
  changed_when: false

- name: Get machines in the machineset after provisioning
  command: >
    oc get machine
    --kubeconfig={{ kubeconfig }}
    --namespace=openshift-machine-api
    --selector='machine.openshift.io/cluster-api-machineset={{ machineset_name }}'
    --output=json
  register: oc_get_machine
  changed_when: false

- name: Add hostname to new_workers_list
  set_fact:
    new_workers_list: "{{ new_workers_list + [ item.status.addresses | selectattr('type', 'match', '^InternalDNS$') | map(attribute='address') | first ] }}"
  loop: "{{ (oc_get_machine.stdout | from_json)['items'] }}"
```

##### EXPECTED RESULTS

The machine(s) should be created without issue.

##### ACTUAL RESULTS

```paste below
The full traceback is:
Traceback (most recent call last):
  File "/home/bbarbach/.ansible/tmp/ansible-tmp-1686938477.3402793-222648-8309146315896/AnsiballZ_azure_rm_virtualmachine_info.py", line 107, in <module>
    _ansiballz_main()
  File "/home/bbarbach/.ansible/tmp/ansible-tmp-1686938477.3402793-222648-8309146315896/AnsiballZ_azure_rm_virtualmachine_info.py", line 99, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/home/bbarbach/.ansible/tmp/ansible-tmp-1686938477.3402793-222648-8309146315896/AnsiballZ_azure_rm_virtualmachine_info.py", line 47, in invoke_module
    runpy.run_module(mod_name='ansible_collections.azure.azcollection.plugins.modules.azure_rm_virtualmachine_info', init_globals=dict(_module_fqn='ansible_collections.azure.azcollection.plugins.modules.azure_rm_virtualmachine_info', _modlib_path=modlib_path),
  File "/usr/lib64/python3.9/runpy.py", line 210, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/lib64/python3.9/runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/usr/lib64/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload_2e91l54l/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_virtualmachine_info.py", line 542, in <module>
  File "/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload_2e91l54l/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_virtualmachine_info.py", line 538, in main
  File "/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload_2e91l54l/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_virtualmachine_info.py", line 350, in __init__
  File "/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload_2e91l54l/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload.zip/ansible_collections/azure/azcollection/plugins/module_utils/azure_rm_common.py", line 472, in __init__
  File "/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload_2e91l54l/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_virtualmachine_info.py", line 368, in exec_module
  File "/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload_2e91l54l/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_virtualmachine_info.py", line 396, in list_items_by_resourcegroup
  File "/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload_2e91l54l/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_virtualmachine_info.py", line 420, in get_vm
  File "/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload_2e91l54l/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_virtualmachine_info.py", line 468, in serialize_vm
UnboundLocalError: local variable 'display_status' referenced before assignment
fatal: [localhost]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/home/bbarbach/.ansible/tmp/ansible-tmp-1686938477.3402793-222648-8309146315896/AnsiballZ_azure_rm_virtualmachine_info.py\", line 107, in <module>\n    _ansiballz_main()\n  File \"/home/bbarbach/.ansible/tmp/ansible-tmp-1686938477.3402793-222648-8309146315896/AnsiballZ_azure_rm_virtualmachine_info.py\", line 99, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/bbarbach/.ansible/tmp/ansible-tmp-1686938477.3402793-222648-8309146315896/AnsiballZ_azure_rm_virtualmachine_info.py\", line 47, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.azure.azcollection.plugins.modules.azure_rm_virtualmachine_info', init_globals=dict(_module_fqn='ansible_collections.azure.azcollection.plugins.modules.azure_rm_virtualmachine_info', _modlib_path=modlib_path),\n  File \"/usr/lib64/python3.9/runpy.py\", line 210, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib64/python3.9/runpy.py\", line 97, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/lib64/python3.9/runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload_2e91l54l/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_virtualmachine_info.py\", line 542, in <module>\n  File \"/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload_2e91l54l/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_virtualmachine_info.py\", line 538, in main\n  File \"/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload_2e91l54l/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_virtualmachine_info.py\", line 350, in __init__\n  File \"/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload_2e91l54l/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload.zip/ansible_collections/azure/azcollection/plugins/module_utils/azure_rm_common.py\", line 472, in __init__\n  File \"/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload_2e91l54l/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_virtualmachine_info.py\", line 368, in exec_module\n  File \"/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload_2e91l54l/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_virtualmachine_info.py\", line 396, in list_items_by_resourcegroup\n  File \"/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload_2e91l54l/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_virtualmachine_info.py\", line 420, in get_vm\n  File \"/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload_2e91l54l/ansible_azure.azcollection.azure_rm_virtualmachine_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_virtualmachine_info.py\", line 468, in serialize_vm\nUnboundLocalError: local variable 'display_status' referenced before assignment\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}

```

**Comment by barbacbd** [comment]
It appears that the issue is in azure_rm_virtualmachine_info.py. Line 468 where the display status may not have been set in the loop if/else statements above as not all cases are covered. 

```
new_result['display_status'] = display_status
```

Machines go into the provisioning state first, but the loop states that it should be in a provisioning state and `code[1]` is failed. before the display status would change. I only mention this case in the loop because it is probably the closest state to what I am in.

**Comment by barbacbd** [comment]
Someone will probably ask for versions so posting here just in case:

```
Package                           Version
--------------------------------- --------
adal                              1.2.7
ansible-core                      2.15.0
applicationinsights               0.11.10
argcomplete                       1.12.3
azure-cli                         2.45.0
azure-cli-core                    2.45.0
azure-cli-telemetry               1.0.8
azure-common                      1.1.11
azure-containerregistry           1.0.0
azure-core                        1.27.1
azure-graphrbac                   0.61.1
azure-identity                    1.7.0
azure-keyvault                    1.1.0
azure-mgmt-apimanagement          3.0.0
azure-mgmt-authorization          2.0.0
azure-mgmt-automation             1.0.0
azure-mgmt-batch                  5.0.1
azure-mgmt-cdn                    11.0.0
azure-mgmt-compute                26.1.0
azure-mgmt-containerinstance      9.0.0
azure-mgmt-containerregistry      9.1.0
azure-mgmt-containerservice       20.0.0
azure-mgmt-core                   1.3.0
azure-mgmt-cosmosdb               6.4.0
azure-mgmt-datafactory            2.0.0
azure-mgmt-datalake-store         1.0.0
azure-mgmt-devtestlabs            3.0.0
azure-mgmt-dns                    8.0.0
azure-mgmt-eventhub               10.1.0
azure-mgmt-hdinsight              9.0.0
azure-mgmt-iothub                 2.2.0
azure-mgmt-keyvault               10.0.0
azure-mgmt-loganalytics           12.0.0
azure-mgmt-managedservices        6.0.0
azure-mgmt-managementgroups       0.2.0
azure-mgmt-marketplaceordering    0.1.0
azure-mgmt-monitor                3.0.0
azure-mgmt-network                19.1.0
azure-mgmt-notificationhubs       7.0.0
azure-mgmt-nspkg                  2.0.0
azure-mgmt-privatedns             1.0.0
azure-mgmt-rdbms                  10.0.0
azure-mgmt-recoveryservices       2.0.0
azure-mgmt-recoveryservicesbackup 3.0.0
azure-mgmt-redis                  13.0.0
azure-mgmt-resource               21.1.0
azure-mgmt-search                 8.0.0
azure-mgmt-servicebus             7.1.0
azure-mgmt-sql                    3.0.1
azure-mgmt-storage                19.0.0
azure-mgmt-trafficmanager         1.0.0b1
azure-mgmt-web                    6.1.0
azure-nspkg                       2.0.0
azure-storage-blob                12.11.0
```

**Comment by Fred-sun** [comment]
Fixes #1123
